lib/approval_request_listener.rb

```diff
@@ -0,0 +1,67 @@
+require "manageiq-messaging"
+
+class ApprovalRequestListener
+  SERVICE_NAME = "platform.approval".freeze
+  CLIENT_AND_GROUP_REF = "approval-catalog-api-worker".freeze
+  EVENT_REQUEST_FINISHED = 'request_finished'.freeze
+
+  attr_accessor :messaging_client_options, :client
+
+  def initialize(messaging_client_options = {})
+    self.messaging_client_options = default_messaging_options.merge(messaging_client_options)
+  end
+
+  def run
+    Thread.new { subscribe_to_approval_updates }
+  end
+
+  def subscribe_to_approval_updates
+    self.client = ManageIQ::Messaging::Client.open(messaging_client_options)
+
+    client.subscribe_topic(
+      :service   => SERVICE_NAME,
+      :max_bytes => 500_000
+    ) do |messages|
+      messages.each do |msg|
+        process_message(msg)
+      end
+    end
+  ensure
+    client&.close
+    self.client = nil
+  end
+
+  private
+
+  def process_message(msg)
+    ProgressMessage.create!(
```

Comment:
Is there any reason why *this* progress message is a `create!` over `create`? Should we be using `create!` in other places?

Reply:
It only is `create!` so we won't get a false positive if there is a problem saving to the db.  The exception that is raised would also stop execution.